src/main/java/tds/testpackageconverter/converter/mappers/TestPackageBlueprintMapper.java

```diff
@@ -289,28 +321,38 @@ private static BlueprintElement mapAssessmentBlueprintElements(final Testspecifi
                 })
                 .collect(Collectors.toCollection(LinkedHashSet::new));
 
-        claimBlueprintElements.forEach(blueprintElement -> mapBlueprintElement(blueprintElement, parentIdToBpElement, scoringMap));
+        claimBlueprintElements.forEach(blueprintElement ->
+                mapBlueprintElement(blueprintElement, parentIdToBpElement, scoringMap, csidEnhancer));
 
         return new ArrayList<>(claimBlueprintElements);
     }
 
     // Recursively map all blueprint elements breadth-first
     private static void mapBlueprintElement(final BlueprintElement currentEl, final Map<String, List<Bpelement>> parentMap,
-                                            final Map<String, Scoring> scoringMap) {
+                                            final Map<String, Scoring> scoringMap, CsidEnhancer csidEnhancer) {
         if (parentMap.containsKey(currentEl.getId())) {
             List<BlueprintElement> blueprintElements = currentEl.blueprintElements();
             List<Bpelement> bpEl = parentMap.get(currentEl.getId());
 
             bpEl.forEach(childEl -> {
                 BlueprintElement currBpEl = BlueprintElement.builder()
                         .setId(childEl.getIdentifier().getName())
+                        .setLabel(csidEnhancer.enhance(childEl.getIdentifier().getName()))
                         .setType(BlueprintElementTypes.TARGET)
                         .setBlueprintElements(new ArrayList<>())
                         .setScoring(Optional.ofNullable(scoringMap.get(childEl.getIdentifier().getUniqueid())))
                         .build();
                 blueprintElements.add(currBpEl);
-                mapBlueprintElement(currBpEl, parentMap, scoringMap);
+                mapBlueprintElement(currBpEl, parentMap, scoringMap, csidEnhancer);
             });
         }
     }
+
+    private static String findSingleProperty(final List<tds.testpackage.legacy.model.Property> properties, final String propertyName) {
+        return properties.stream()
+                .filter(property -> property.getName().equalsIgnoreCase(propertyName))
+                .map(tds.testpackage.legacy.model.Property::getValue)
+                .findAny()
+                .orElseThrow(() -> new IllegalArgumentException("No subject property was found in the test specification package"));
```

Comment:
I think the message within this `IllegalArgumentException` should use the propertyName passed into the method.  It is currently hardcoded to `subject` even though the property could be `grade.